### PR TITLE
Cleanup eslint errors

### PR DESCRIPTION
**Pull request by mdurling**
Now that we are ~100% typescript we can clean up the ESLint config as we don't need to worry about linting config files and such. Also, resolve all linter warnings and turn off the `only-warn` plugin.

### CODE REVIEW

**Comment by avatarneil** [comment]
Mr. Jenkins! Please run tests

**Comment by mdurling** [comment]
Please run tests again

File: packages/mds-agency/api.ts [hunk 1]
```diff
@@ -1154,8 +1165,8 @@ function api(app: express.Express): express.Express {
         return map
       }, eventSeed)
       const telemetrySeed: { [s: string]: Telemetry } = {}
-      const telemetryMap = telemetry.reduce((map, telemetry) => {
-        map[telemetry.device_id] = telemetry
+      const telemetryMap = telemetry.reduce((map, t) => {
+        map[t.device_id] = t
```
Comment (avatarneil) [inline packages/mds-agency/api.ts]: Could this be refactored using `Object.assign()`? Something like:
```typescript
const telemetryMap = telemetry.reduce((map, t) => {
   return Object.assign(map, { [t.device_id]: t })
}
```

Comment (mdurling) [reply packages/mds-agency/api.ts]: Yes. And in fact, we do this so often that we should create a helper function.

**Comment by avatarneil** [comment]
Mr. Jenkinsssss, bring me a dream (please rerun tests) 
Make this the best PR that I've ever seen!

**Comment by mdurling** [comment]
Run tests with updated Jenkins build steps

**Pull request merged**